doc/sphinx/Pacemaker_Administration/options.rst

```diff
@@ -0,0 +1,233 @@
+       details.
+   * - .. _EDITOR:
+
+       .. index::
+          single: EDITOR
+          single: environment variable; EDITOR
+
+       EDITOR
+     -
+     - Text editor to use for editing shadow files. Required for the ``--edit``
+       command of :ref:`crm_shadow <crm_shadow>`.
+   * - .. _HOME:
+
+       .. index::
+          single: HOME
+          single: environment variable; HOME
+
+       HOME
+     - (see description)
+     - Current user's home directory. Used to create a default
+       :ref:`CIB_shadow_dir <CIB_shadow_dir>` for non-privileged users.
+       Defaults to the user's home directory as configured in the passwd
+       database, if an entry exists.
+   * - .. _OCF_RESKEY_CRM_meta_on_node:
+
+       .. index::
+          single: OCF_RESKEY_CRM_meta_on_node
+          single: environment variable; OCF_RESKEY_CRM_meta_on_node
+
+       OCF_RESKEY_CRM_meta_on_node
+     -
+     - Advanced use only: If
+       :ref:`CRM_meta_physical_host <CRM_meta_physical_host>` is not set, or
+       if :ref:`CRM_meta_container_attribute_target <CRM_meta_container_attribute_target>`
+       is not set to ``host``, then Pacemaker uses this as the default node name
+       when setting attributes. Pacemaker sets this variable based on the
+       local node name before executing a resource agent, so that it's present
+       in the environment when a resource agent requests an attribute update.
+   * - .. _PE_fail:
+
+       .. index::
+          single: PE_fail
+          single: environment variable; PE_fail
+
+       PE_fail
+     - 0
+     - Advanced use only: A dummy graph action with action ID matching this
+       option will be marked as failed. Primarily for developer use with
+       scheduler simulations.
+   * - .. _PS1:
+
+       .. index::
+          single: PS1
+          single: environment variable; PS1
+
+       PS1
+     -
+     - The shell's primary prompt string. Used by
+       :ref:`crm_shadow <crm_shadow>`: set to indicate that the user is in an
+       interactive shadow CIB session, and checked to determine whether the user
+       is already in an interactive session before creating a new one.
+   * - .. _SHELL:
+
+       .. index::
+          single: SHELL
+          single: environment variable; SHELL
+
+       SHELL
+     -
+     - Absolute path to a shell. Used by :ref:`crm_shadow <crm_shadow>` when
+       launching an interactive session.
+   * - .. _TMPDIR:
+
+       .. index::
+          single: TMPDIR
+          single: environment variable; TMPDIR
+
+       TMPDIR
+     - /tmp
+     - Directory for temporary files. If not an absolute path, the default is
+       used instead.
+
+.. _PCMK_service:
+
+Additionally, Pacemaker sets the ``PCMK_service`` environment variable to the
+subsystem or tool name. External scripts such as resource agents and fence
+agents can check this variable to see whether they're being run by (for example)
+the executor or the fencer.
```

Comment:
Let's move PCMK_service, CRM_meta_container_attribute_target, CRM_meta_physical_host, and OCF_RESKEY_CRM_meta_on_node into a separate new table of variables set by Pacemaker for OCF resource agent use, in the OCF resource agents section. We can add a blanket warning that users should not set any of these, and if they do, behavior is undefined; that Pacemaker may define other variables, but their names, values, and meanings are subject to change; and that non-OCF agents should not rely on any of these being set.

We can drop the `OCF_RESKEY_ip` example from Pacemaker Explained, and instead add an entry for `OCF_RESKEY_*` to the new table. We can also move all other references to `OCF_RESKEY` (clone variables etc.) from Pacemaker Explained.

For PCMK_service, let's say that the values are subject to change, and so useful mainly for logging (i.e. don't say they should check the variable).

That will cleanly separate cluster configuration and agent development, and give agent developers everything they need in one place.

Reply:
Bah, I didn't know that. For their purposes, `!= ""` would be sufficient, and better than `== "pacemaker-fenced"`.

Reply:
That depends. Anything that calls `pcmk__cli_init_logging()`, `crm_log_init()`, or `crm_log_preinit()` with a non-NULL first argument gets `PCMK_service` set (to `crm_system_name`). That includes most or all Pacemaker CLI tools but seemingly *not* the CTS helpers. (I haven't run them to make sure `PCMK_service` is unset.)

If checking whether the fence agent is being called by *any* Pacemaker tool (besides `cts-fence-helper`) is sufficient, then checking for non-empty is sufficient. If the fencing library wants to know whether it's called by a daemon, I'm not sure if there's something else it can check.

Reply:
> We can also move all other references to `OCF_RESKEY` (clone variables etc.) from Pacemaker Explained.

There are 116 of them. We could move the entire section 11.2.9 ("Clone Resource Agent Requirements") from Pacemaker Explained to Pacemaker Administration.

The division of resource documentation into Administration and Explained feels more arbitrary than usual, and I struggle to draw lines between them. I guess, very loosely, info that's more relevant to **writing resource agents** (e.g., what environment variables are exposed) can go into Administration, while info pertaining to generic resource **behavior** and **configuration** can go into Explained.

Reply:
> We could move the entire section 11.2.9 ("Clone Resource Agent Requirements") from Pacemaker Explained to Pacemaker Administration.

I think that's the way to go.